### ISSUE TITLE
Uncaught TypeError: Cannot read property 'tags' of undefined

### ISSUE THREAD

**Issue by henrytseng** [samu/blocky#20]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Mac OS X 10.12.2
**Thrown From**: [blocky](https://github.com/samu/blocky) package, v0.10.0

### Stack Trace

Uncaught TypeError: Cannot read property 'tags' of undefined

```
At /Users/henry/.atom/packages/blocky/lib/blockmap-compiler.coffee:88

TypeError: Cannot read property 'tags' of undefined
    at module.exports (/Users/henry/.atom/packages/blocky/lib/blockmap-compiler.coffee:88:16)
    at BlockyView.module.exports.BlockyView.notifyContentsModified (/Users/henry/.atom/packages/blocky/lib/blocky-view.coffee:31:17)
    at /Users/henry/.atom/packages/blocky/lib/blocky-view.coffee:11:54
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1417:25
```

### Commands

```
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig",
        "BUCK.autodeps"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "destroyEmptyPanes": false,
    "disabledPackages": [
      "linter-swagger"
    ],
    "telemetryConsent": "limited"
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.16 (inactive)
atom-eco, v0.2.0 (active)
atom-jade, v0.3.0 (active)
autoclose-html, v0.23.0 (active)
blocky, v0.10.0 (active)
docblockr, v0.9.1 (active)
gitkraken, v0.4.2 (active)
incremental-search, v5.2.1 (inactive)
language-babel, v2.49.4 (active)
language-haml, v0.24.1 (active)
language-hl7, v0.1.1 (active)
language-scala, v1.1.8 (active)
linter, v1.11.21 (active)
linter-eslint, v8.0.0 (active)
linter-rubocop, v0.5.2 (active)
linter-scalac, v1.4.2 (active)
linter-swagger, v0.3.8 (inactive)
scala-format, v0.1.8 (active)
split-diff, v1.1.1 (inactive)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (active)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (active)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.33.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.43.2 (active)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (active)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.5 (active)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by tyrollins** [comment]
Just ran into this same error...

**Comment by samu** [comment]
Hey Tyler, 
do you know how to reproduce this?

**Comment by tyrollins** [comment]
I'm unable to consistently reproduce the error, it is a bit random. I have noticed however that this happens upon start up / opening my first ruby file. 

Additionally, the block highlighting appears fine, even after error is raised. 

Not sure if this helps but here is the Error & Stack Trace:
```
Uncaught TypeError: Cannot read property 'tags' of undefined
```
```
/Users/username/.atom/packages/blocky/lib/blockmap-compiler.coffee:88
```
```
TypeError: Cannot read property 'tags' of undefined
    at module.exports (/Users/username/.atom/packages/blocky/lib/blockmap-compiler.coffee:88:16)
    at BlockyView.module.exports.BlockyView.notifyContentsModified (/Users/username/.atom/packages/blocky/lib/blocky-view.coffee:31:17)
    at /Users/username/.atom/packages/blocky/lib/blocky-view.coffee:11:54
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1385:25
```

**Comment by shadowbq** [comment]
Here is mine:

happened upon start up / opening my first ruby file

## Uncaught TypeError: Cannot read property 'tags' of undefined

```
packages/blocky/lib/blockmap-compiler.coffee:88
```

```
TypeError: Cannot read property 'tags' of undefined
    at module.exports (/home/username/.atom/packages/blocky/lib/blockmap-compiler.coffee:88:16)
    at BlockyView.module.exports.BlockyView.notifyContentsModified (/home/username/.atom/packages/blocky/lib/blocky-view.coffee:31:17)
    at /home/username/.atom/packages/blocky/lib/blocky-view.coffee:11:54
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at /usr/share/atom/resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1371:25
```

**Comment by FunkyloverOne** [comment]
For me it usually happens while switching git branches.

**Comment by BluGeni** [comment]
I had this happen when running 'rails db:migrate'

**Comment by samu** [comment]
fixed in `v0.11.0`
